Type declarations for DropDown.d.ts and NavDropDown.d.ts have invalid import to @restart/ui/esm/types

### Prerequisites

- [X] I am using the [correct version](https://github.com/react-bootstrap/react-bootstrap#bootstrap-compatibility) of React-Bootstrap for my version of Bootstrap
- [X] I have [searched](https://github.com/react-bootstrap/react-bootstrap/issues?q=is%3Aissue) for duplicate or closed issues
- [X] I have read the [contributing guidelines](https://github.com/react-bootstrap/react-bootstrap/blob/master/CONTRIBUTING.md)

### Describe the bug

When referencing react-bootstrap from a typescript project that has tsconfig options ("moduleResolution": "bundler" OR "resolvePackageJsonExports": true) AND "skipLibCheck": false, the tsc build fails with the following error:

```
node_modules/react-bootstrap/esm/Dropdown.d.ts:16:49 - error TS2307: Cannot find module '@restart/ui/esm/types' or its corresponding type declarations.

16     Item: BsPrefixRefForwardingComponent<import("@restart/ui/esm/types").DynamicRefForwardingComponent<React.ForwardRefExoticComponent<import("@restart/ui/esm/Button").ButtonProps & React.RefAttributes<HTMLElement>>, import("@restart/ui/Dropdown").DropdownItemProps>, import("./DropdownItem").DropdownItemProps>;
                                                   ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/react-bootstrap/esm/NavDropdown.d.ts:15:49 - error TS2307: Cannot find module '@restart/ui/esm/types' or its corresponding type declarations.

15     Item: BsPrefixRefForwardingComponent<import("@restart/ui/esm/types").DynamicRefForwardingComponent<React.ForwardRefExoticComponent<import("@restart/ui/esm/Button").ButtonProps & React.RefAttributes<HTMLElement>>, import("@restart/ui/esm/DropdownItem").DropdownItemProps>, import("./DropdownItem").DropdownItemProps>;
                                                   ~~~~~~~~~~~~~~~~~~~~~~~


Found 2 errors in 2 files.
```

Since this is an error in the package type declarations, there is nothing the user can do to fix it aside from turning skipLibCheck to false which really just masks the problem.

The problem comes from the generated type declaration importing from "@restart/ui/esm/DropdownItem" when it should be "@restart/ui/DropdownItem" 

### Expected behavior

Expect react-bootstrap type declaration files to use correct imports without esm segment

"@restart/ui/Button"
"@restart/ui/types"

instead of 

"@restart/ui/esm/Button"
"@restart/ui/esm/types"

### To Reproduce

Create a new react typescript app (here is example using yarn/vite)
1. `yarn create vite vite-test --template react-ts`
2. `yarn`
3. `yarn add react-bootstrap`
4. Edit tsconfig.json, change skipLibCheck to false
5. Edit src/App.tsx to use any react-bootstrap component `<Button>Test</Button>`
6. `yarn build`
7. You should see the following error
```
node_modules/react-bootstrap/esm/Dropdown.d.ts:16:49 - error TS2307: Cannot find module '@restart/ui/esm/types' or its corresponding type declarations.

16     Item: BsPrefixRefForwardingComponent<import("@restart/ui/esm/types").DynamicRefForwardingComponent<React.ForwardRefExoticComponent<import("@restart/ui/esm/Button").ButtonProps & React.RefAttributes<HTMLElement>>, import("@restart/ui/Dropdown").DropdownItemProps>, import("./DropdownItem").DropdownItemProps>;
                                                   ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/react-bootstrap/esm/NavDropdown.d.ts:15:49 - error TS2307: Cannot find module '@restart/ui/esm/types' or its corresponding type declarations.

15     Item: BsPrefixRefForwardingComponent<import("@restart/ui/esm/types").DynamicRefForwardingComponent<React.ForwardRefExoticComponent<import("@restart/ui/esm/Button").ButtonProps & React.RefAttributes<HTMLElement>>, import("@restart/ui/esm/DropdownItem").DropdownItemProps>, import("./DropdownItem").DropdownItemProps>;
                                                   ~~~~~~~~~~~~~~~~~~~~~~~


Found 2 errors in 2 files.
```

### Reproducible Example

https://codesandbox.io/s/withered-moon-htt967

### Screenshots

_No response_

### What operating system(s) are you seeing the problem on?

_No response_

### What browser(s) are you seeing the problem on?

_No response_

### What version of React-Bootstrap are you using?

2.8.0

### What version of Bootstrap are you using?

5.0

### Additional context

_No response_

--- a/src/DropdownItem.tsx
+++ b/src/DropdownItem.tsx
@@ -1,7 +1,7 @@
 import classNames from 'classnames';
 import PropTypes from 'prop-types';
 import * as React from 'react';
-import BaseDropdownItem, {
+import {
   useDropdownItem,
   DropdownItemProps as BaseDropdownItemProps,
 } from '@restart/ui/DropdownItem';
@@ -45,47 +45,45 @@ const propTypes = {
   as: PropTypes.elementType,
 };
 
-const DropdownItem: BsPrefixRefForwardingComponent<
-  typeof BaseDropdownItem,
-  DropdownItemProps
-> = React.forwardRef(
-  (
-    {
-      bsPrefix,
-      className,
-      eventKey,
-      disabled = false,
-      onClick,
-      active,
-      as: Component = Anchor,
-      ...props
-    },
-    ref,
-  ) => {
-    const prefix = useBootstrapPrefix(bsPrefix, 'dropdown-item');
-    const [dropdownItemProps, meta] = useDropdownItem({
-      key: eventKey,
-      href: props.href,
-      disabled,
-      onClick,
-      active,
-    });
+const DropdownItem: BsPrefixRefForwardingComponent<'a', DropdownItemProps> =
+  React.forwardRef(
+    (
+      {
+        bsPrefix,
+        className,
+        eventKey,
+        disabled = false,
+        onClick,
+        active,
+        as: Component = Anchor,
+        ...props
+      },
+      ref,
+    ) => {
+      const prefix = useBootstrapPrefix(bsPrefix, 'dropdown-item');
+      const [dropdownItemProps, meta] = useDropdownItem({
+        key: eventKey,
+        href: props.href,
+        disabled,
+        onClick,
+        active,
+      });
 
-    return (
-      <Component
-        {...props}
-        {...dropdownItemProps}
-        ref={ref}
-        className={classNames(
-          className,
-          prefix,
-          meta.isActive && 'active',
-          disabled && 'disabled',
-        )}
-      />
-    );
-  },
-);
+      return (
+        <Component
+          {...props}
+          {...dropdownItemProps}
+          ref={ref}
+          className={classNames(
+            className,
+            prefix,
+            meta.isActive && 'active',
+            disabled && 'disabled',
+          )}
+        />
+      );
+    },
+  );
 
 DropdownItem.displayName = 'DropdownItem';
 DropdownItem.propTypes = propTypes;

